Backend/src/main/java/com/qelem/api/controller/UserRestController.java

```diff
@@ -65,9 +81,24 @@ public UserDto userById(@PathVariable("id") Long id) {
         return new UserDto(userModel);
     }
 
+    @PatchMapping(path = "/changePassword/{id}", consumes = "application/json")
+    public UserModel changePassword(@PathVariable("id") Long id,
+                                    @RequestBody ChangePasswordModel user) {
+
+        UserModel userModel = userRepository.findById(id).get();
+
+        if (passwordEncoder.encode(user.getOldPassword()).matches(userModel.getPassword())) {
+            throw new PasswordException("Password doesn't match!");
+        }
+        userModel.setPassword(passwordEncoder.encode(user.getNewPassword()));
+        UserModel res = userRepository.save(userModel);
+        res.setPassword("");
+        return res;
+    }
+
     @PatchMapping(path = "/{id}", consumes = "application/json")
     public UserModel updateUser(@PathVariable("id") Long id,
-            @RequestBody UserModel user) {
+                                @RequestBody ChangePasswordModel user) {
         var isAuthorized = loggedInUser().getId().equals(id) || loggedInUser().isAdmin();
 
         if (!isAuthorized) {
```

Comment:
What is this?, we already have an endpoint to change password eko.

Reply:
This happened because of the major changes you made on the API